javatests/arcs/core/storage/keys/JoinStorageKeyTest.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.storage.keys
+
+import arcs.core.storage.referencemode.embed
+import arcs.core.storage.StorageKeyParser
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/** Tests for [JoinStorageKey]. */
+@RunWith(JUnit4::class)
+class JoinStorageKeyTest {
+    @Test
+    fun toString_rendersCorrectly() {
+        val key1 = RamDiskStorageKey("key1")
+        val key2 = RamDiskStorageKey("key2")
+        val key = JoinStorageKey(2, listOf(key1, key2))
+
+        assertThat(key.toString())
+            .isEqualTo("$COMPOSITE_PROTOCOL://2/{$key1}{$key2}")
```

Comment:
while it's good to have a constant (i.e. COMPOSITE_PROTOCOL) it would also be good to actually test what we get at the end, i.e. `join://...`, can we inline the constant at least in this test?